classes/class-ledyer-lco-gateway.php

```diff
@@ -242,6 +242,29 @@ public function enqueue_scripts() {
 		public function process_payment( $order_id ) {
 			$order = wc_get_order( $order_id );
 
+			// HPP Redirect flow.
+			if ( 'redirect' === ( $this->settings['checkout_flow'] ?? 'embedded' ) ) {
+				$data         = \Ledyer\Requests\Helpers\Woocommerce_Bridge::get_cart_data();
+				$ledyer_order = ledyer()->api->create_order_session( $data, $order_id );
+				lco_create_or_update_order();
```

Comment:
Is the `create_order_session` here necessary? I would think that if:

```
is_wc_endpoint_url( 'order-pay' ) || 'redirect' === ( $this->settings['checkout_flow'] ?? 'embedded' )
```

Then you could just call the `process_direct_handler` directly.